modin/pandas/dataframe.py

```diff
@@ -464,16 +464,18 @@ def groupby(
             else:
                 mismatch = len(by) != len(self.axes[axis])
                 if mismatch and all(
-                    obj in self
-                    or (hasattr(self.index, "names") and obj in self.index.names)
+                    isinstance(obj, str)
+                    and (
+                        obj in self
+                        or (hasattr(self.index, "names") and obj in self.index.names)
+                    )
                     for obj in by
                 ):
                     # In the future, we will need to add logic to handle this, but for now
                     # we default to pandas in this case.
                     pass
-                elif mismatch:
+                elif mismatch and all(isinstance(obj, str) for obj in by):
```

Comment:
Will this raise a `KeyError` if `by` is a list like that `["column_that_not_in_df", some_series]`?

Reply:
Wrote a test, working through it, and ugh, Pandas is inconsistent:

```python
(Pdb) p pandas_df
    b    c
0  11  111
1  11  111
2  22  222
(Pdb) pandas_df.groupby(by=["a", "b", "c"])
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x7f351c17e670>
(Pdb) pandas_df.groupby(by=["a", "b"])
*** KeyError: 'a'
```

Reply:
I think the 3-length cases doesn't raise because it happens to be same length as the index. Not a good API.

Reply:
OK, done.